javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/typesystem/ReferenceTypeImpl.java

```diff
@@ -203,13 +204,36 @@ public ResolvedType transformTypeParameters(ResolvedTypeTransformer transformer)
                 .collect(Collectors.toList());
 
         // Avoid repetitions of Object
-        ancestors.removeIf(a -> a.getQualifiedName().equals(Object.class.getCanonicalName()));
-        ResolvedReferenceTypeDeclaration objectType = typeSolver.solveType(Object.class.getCanonicalName());
-        ResolvedReferenceType objectRef = create(objectType);
-        ancestors.add(objectRef);
+        ancestors.removeIf(ResolvedReferenceType::isJavaLangObject);
+        ResolvedReferenceTypeDeclaration objectType = typeSolver.getSolvedJavaLangObject();
+        ancestors.add(create(objectType));
+
         return ancestors;
     }
 
+    private boolean hasSuperClassDifferentToSelf(ResolvedReferenceTypeDeclaration typeDeclaration) {
+        if(!typeDeclaration.isClass()) {
+            // If self is not a class, there is no super class
```

Comment:
hmm... is this actually true, given that interfaces are able to extend other interfaces?

... but does this create a "superClass" hierarchy?

I'm leaning towards leaving it as-is since, if we do consider this, a list of "super" interfaces will be required as it would allow multiple "parents"...

Reply:
Either way, this comment could be expanded to make this thought process more explicit for the benefit of anybody coming across this again in the future.

Reply:
Hmm.. something to consider here is that enums implicitly extend `java.lang.Enum`...

Perhaps additional tests are required to figure out the current behaviour versus expected behaviour.

Reply:
This is an unused new private method, thus can be resolved by removing it...

I still believe that this would be helpful to introduce on the root nodes (ResolvedReferenceType and ResolvedReferenceTypeDeclaration) to avoid the MANY repeated checks against java.lang.Object... but that can be added within a separate PR...